src/App.tsx

```diff
@@ -41,65 +50,99 @@ interface IRepositoryApiClientExInternal extends IRepositoryApiClientEx {
   _repoName?: string;
 }
 
-export default class App extends React.Component<{}, { expandFolderBrowser: boolean, selectedFolder: LfFolder | undefined, selectedFile?: File; isLoggedIn: boolean }> {
+interface ILfSelectedFolder {
+  selectedNodeUrl: string; // url to open the selected node in Web Client
+  selectedFolderPath: string; // path of selected folder
+  selectedFolderName: string; // name of the selected folder
+}
+
+export default class App extends React.Component<
+any, 
+{ 
+  expandFolderBrowser: boolean;
+  lfSelectedFolder?: ILfSelectedFolder; 
+  selectedFile?: File; 
+  isLoggedIn: boolean; 
+  shouldShowOpen: boolean; 
+  shouldShowSelect: boolean; 
+  shouldDisableSelect: boolean; 
+  show: boolean;
+  popupErrorMessage: string;
+}> {
   REDIRECT_URI: string = 'REPLACE_WITH_YOUR_REDIRECT_URI'; // i.e http://localhost:3000, https://serverName/lf-sample/index.html
   CLIENT_ID: string = 'REPLACE_WITH_YOUR_CLIENT_ID';
   HOST_NAME: string = ''; // only add this if you are using a different region or environment (i.e. laserfiche.ca, eu.laserfiche.com)
-  REGIONAL_DOMAIN: string = 'laserfiche.com' // only update this if you are using a different region or environment
 
   loginComponent: React.RefObject<NgElement & WithProperties<LfLoginComponent>>;
-  folderBrowser: React.RefObject<NgElement & WithProperties<LfFileExplorerComponent>>;
+  repositoryBrowser: React.RefObject<NgElement & WithProperties<LfRepositoryBrowserComponent>>;
   repoClient?: IRepositoryApiClientExInternal;
   creationDate?: string;
-  lfRepoTreeService?: LfRepoTreeService;
-  selectedNodeUrl: string | undefined;
-  selectedFolderDisplayName: string | undefined;
+  lfRepoTreeService?: LfRepoTreeNodeService;
   fileName: string | undefined;
   fileExtension: string | undefined;
   fileInput: React.RefObject<HTMLInputElement> | undefined;
   fieldContainer: React.RefObject<NgElement & WithProperties<LfFieldContainerComponent>>;
   fieldsService?: LfFieldsService;
-  localizationService = new LfLocalizationService(resources);
+  localizationService: LfLocalizationService = new LfLocalizationService(resources);
+  entrySelected: LfRepoTreeNode | undefined;
+  toolBarElement: React.RefObject<NgElement & WithProperties<LfToolbarComponent>>;
 
   constructor(props: any) {
     super(props);
     this.loginComponent = React.createRef();
-    this.folderBrowser = React.createRef();
+    this.repositoryBrowser = React.createRef();
     this.fileInput = React.createRef();
     this.fieldContainer = React.createRef();
-    this.setState(() => { return { expandFolderBrowser: false, isLoggedIn: false, selectedFolder: undefined } });
+    this.toolBarElement = React.createRef();
+    this.setState({
+      expandFolderBrowser: false, 
```

Comment:
is this version of setState working for you? I had to use the lambda version

Reply:
that is weird. this setState works for me